data-pipeline-flink/user-cache-updater/src/main/scala/org/sunbird/dp/usercache/functions/UserCacheUpdaterFunction.scala

```diff
@@ -152,9 +152,121 @@ class UserCacheUpdaterFunction(config: UserCacheUpdaterConfig)(implicit val mapT
       record.getString("type").toLowerCase match {
         case config.stateKey => result.put(config.stateKey, record.getString("name"))
         case config.districtKey => result.put(config.districtKey, record.getString("name"))
+        case config.blockKey => result.put(config.blockKey, record.getString("name"))
       }
     })
     result
   }
 
+  /**
+    * It Will return the rootOrgId to filter out custodian users and state users
+    * @param metrics
+    * @return custodianRootOrgId
+    */
+  def getCustodianRootOrgId(metrics: Metrics): String = {
+    val custRootOrgId = readFromCassandra(
+      keyspace = config.keySpace,
+      table = config.systemSettingsTable,
+      QueryBuilder.eq("id", "custodianRootOrgId"),
+      metrics)
+    custRootOrgId.get(0).getString(2)
+  }
+
+  /**
+    *
+    * @param userDetails, custRootOrgId
+    * @return Map having isCustodianUser as true/false
+    */
+  def getIsCustodianUser(userDetails: mutable.Map[String, AnyRef], custRootOrgId: String): mutable.Map[String, AnyRef] ={
```

Comment:
I am assuming the function is used to check if a user belongs to a custodian org. The function should be named appropriately. Also, we shouldn't be passing around a mutable Map to add a boolean value to the Map. The value needs to be updated in the calling function.

Reply:
Done